### PR TITLE
Comparison bars

### PR DESCRIPTION
**Pull request by drakonen**
Added Bar vis

@Juice10 @kaareal @kevinvandijk

### CODE REVIEW

File: docs/examples/visualizations/bar.html [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+              ],
+              label: 'previous',
+              period: prevPeriod
+            }
+          ],
+            labels: ['first', '<b>bold</b>', 'third', '4th',  '5th', '6th']
+          };
+
+          var facebookImg = '/images/FB-f-Logo__blue_50.png';
+          var twitterImg = '/images/Twitter_logo_blue.png';
```
Comment (Juice10) [inline docs/examples/visualizations/bar.html]: those are some ugly ass filenames, could use some renaming.
`twitter-logo-blue.png` and `facebook-logo-blue.png` are a lot nicer

Comment (drakonen) [inline docs/examples/visualizations/bar.html]: Yeah.. this is how they came from their packages.

File: docs/examples/visualizations/bar.html [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+      require(['../../config'], function () {
+        require.config({
+          baseUrl: '/src'
+        });
+
+        // Do not use the concatenated Openhose for dev purposes
+        require.config({
+          paths: {
+//            openhose: undefined
```
Comment (kevinvandijk) [inline docs/examples/visualizations/bar.html]: Leftover?

File: src/processor/bar.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+define(['../core', './analytics-mixin'], function(Openhose, AnalyticsMixin) {
+
+  var processData = function(dimensions, metrics) {
+    var ret = {};
```
Comment (Juice10) [inline src/processor/bar.js]: we should probably rename `ret` to something less cryptic

File: docs/examples/visualizations/bar.html [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                { label: "1.4", value: 3 },
+                { label: "1.5", value: 0 },
+                { label: "1.6", value: 5 }
+              ],
+              label: 'current',
+              period: period
+            }, {
+              values: [
+                { label: "Label 2.1", value: 5 },
+                { label: "2.2", value: 1 },
```
Comment (kevinvandijk) [inline docs/examples/visualizations/bar.html]: Because OCD: *Label 2.2

File: src/styles/rickshaw.less [hunk 5]
```diff
@@ -153,7 +153,7 @@
 }
 
 
-.rickshaw_graph .detail .x_label {
+.rickshaw_graph .detail .x_label, .hover-container {
```
Comment (Juice10) [inline src/styles/rickshaw.less]: I wonder if we should be editing this file...

File: src/processor/analytics-mixin.js [hunk 6]
```diff
@@ -269,26 +269,31 @@ define(['../core', '../lib/utils'], function(Openhose, Utils) {
...
+        var segment = [];
+        dim.collection.forEach(function(summary) {
+          summary = summary.toJSON ? summary.toJSON() : summary;
+          if(this.options.filter && !this.options.filter(summary)) {
+            return false;
+          }
+          var secondaryValues = {};
+          var metricIds = dim.options.metrics;
+
+          for (var i=1; i < metricIds.length; i++) {
```
Comment (kevinvandijk) [inline src/processor/analytics-mixin.js]: Why start from the second element of metricIds?

**Comment by Juice10** [comment]
:+1:

File: src/styles/rickshaw.less (hunk 5 above)
Comment (drakonen) [inline src/styles/rickshaw.less]: I believe we have already edited this file.

File: src/styles/bar.less [hunk 7]
```diff
@@ -0,0 +1,80 @@
+.bar {
+  height: 100%;
+  position: relative;
+
+  .hover-container {
+    color: black;
+    position: absolute;
+    top: 0px;
+    pointer-events: none;
```
Comment (kevinvandijk) [inline src/styles/bar.less]: Not sure if it matters at all here, but be aware that pointer-events aren't supported in IE < 11.

File: docs/examples/visualizations/bar.html [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+              if (i === 0) {
+                // make the first bar narrower
+                var width = this.style.width;
+                width = width.slice(0, -1) * 0.5;
+                this.style.width = width + '%';
+                this.firstChild.style.display = 'none';
+              } else if (i === 1) {
+                // and make the second bar wider
+                var width = this.style.width;
+//                width = width.slice(0, -1) * 1.3;
```
Comment (kevinvandijk) [inline docs/examples/visualizations/bar.html]: Commented code

File: src/models/dimension.js [hunk 9]
```diff
@@ -60,7 +60,8 @@ define([
 
     getParamsV3: function(options) {
       if (!options) options = {};
-      var timeframe = this.get('previousPeriod') ? this.period.timeframePreviousPeriod() : this.period.timeframe();
+      //var timeframe = this.get('previousPeriod') ? this.period.timeframePreviousPeriod() : this.period.timeframe();
```
Comment (kevinvandijk) [inline src/models/dimension.js]: Commented code

File: src/processor/bar.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+    if (dimensions.length) {
+      ret.values = AnalyticsMixin._segmentsForDimension.call(this);
+      //ret.period = dimensions[0].period;
```
Comment (kevinvandijk) [inline src/processor/bar.js]: Commented code

File: src/styles/bar.less [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+      position: absolute;
+      top: 0;
+      left: 0;
+      right: 0;
+      padding-top: 50%; // .bar-image height
+    }
+
+    .bar-image {
+      width: 50%;
+      height: auto;
```
Comment (kevinvandijk) [inline src/styles/bar.less]: Should be unnecessary, auto is the default value

**Comment by kevinvandijk** [comment]
Cool!  :+1:

**Pull request merged**